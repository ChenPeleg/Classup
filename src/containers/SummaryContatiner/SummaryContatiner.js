import React from "react";
import QuestionWrapper from "../../components/QuestionWrapper/QuestionWrapper";
import QuestionText from "../../components/QuestionText/QuestionText";
import SubmitButton from "../../components/SubmitButton/SubmitButton";
import SummaryText from "../../components/SummaryText/SummaryText";
import SummaryTable from "../../components/SummaryTable/SummaryTable";

const test1 = [
  [],
  ["INFO"],
  ["WRONG", "RIGHT"],
  ["RIGHT"],
  ["WRONG", "WRONG", "WRONG", "RIGHT"],
  ["RIGHT"],
  ["RIGHT"],
  ["WRONG", "WRONG", "RIGHT"],
  ["WRONG", "RIGHT"],
  ["RIGHT"],
  ["RIGHT"],
  ["RIGHT"],
  ["WRONG", "RIGHT"],
];
const summaryObject = (allQuestions) => {
  allQuestions = test1;
  let numOfquestions = 0;
  let mistakesObject = { q0: 0, q1: 0, q23: 0 };
  const sumArray = allQuestions.filter((e) => e.includes("RIGHT"));
  for (let i = 0; i < sumArray.length; i++) {
    const wrongs = sumArray[i].filter((e) => e === "WRONG").length;
    const rights = sumArray[i].filter((e) => e === "RIGHT").length;
    let mistakeCound = wrongs < 2 ? "q" + wrongs : "q23";
    mistakesObject[mistakeCound] += 1;
    numOfquestions += rights;
  }

  return { mistakesObject, numOfquestions };
};
const SummaryContainer = (props) => {
  const submitHandler = () => {
    props.resetHandler();
  };
  return (
    <QuestionWrapper>
      <QuestionText>
        <SummaryText>
          <SummaryTable summaryObject={summaryObject(props.sumData)} />
        </SummaryText>
      </QuestionText>
      <SubmitButton submitHandler={submitHandler}>Start Again</SubmitButton>
    </QuestionWrapper>
  );
};
export default SummaryContainer;
