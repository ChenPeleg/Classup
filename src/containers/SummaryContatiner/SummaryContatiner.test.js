import React from "react";
import { shallow } from "enzyme";
import SummaryContatiner from "./SummaryContatiner";

// const test1 = 

const props = {
  sumData: [
    [],
    ["INFO"],
    ["WRONG", "RIGHT"],
    ["RIGHT"],
    ["WRONG", "WRONG", "WRONG", "RIGHT"],
    ["RIGHT"],
    ["RIGHT"],
    ["WRONG", "WRONG", "RIGHT"],
    ["WRONG", "RIGHT"],
    ["RIGHT"],
    ["RIGHT"],
    ["RIGHT"],
    ["WRONG", "RIGHT"],
  ]
};
const wrap = shallow(<SummaryContatiner {...props} />)

describe("<SummaryContatiner>", () => {
  it("renders correctly", () => { expect(wrap).toMatchSnapshot() })
})