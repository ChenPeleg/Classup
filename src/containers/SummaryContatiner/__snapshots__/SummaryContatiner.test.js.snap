// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryContatiner> renders correctly 1`] = `
<QuestionWrapper>
  <QuestionText>
    <SummaryText>
      <SummaryTable
        summaryObject={
          Object {
            "mistakesObject": Object {
              "q0": 6,
              "q1": 3,
              "q23": 2,
            },
            "numOfquestions": 11,
          }
        }
      />
    </SummaryText>
  </QuestionText>
  <SubmitButton
    submitHandler={[Function]}
  >
    Start Again
  </SubmitButton>
</QuestionWrapper>
`;
